File: frontend/app/models/activity.ts
import { Address } from 'app/models/address';
import { Category } from 'app/models/category';
import { Model } from 'app/models/model';
import { Provider } from 'app/models/provider';
import { Tag } from 'app/models/tag';
import { TargetGroup } from 'app/models/target-group';
import { Schedule } from 'app/models/schedule';


export class Activity extends Model {
	constructor(json: any = {} as Activity) {
		super(json ? json.id : '');
		if (json) {
			if (json.name) { this.name = json.name; }
			if (json.description) { this.description = json.description; }
			if (json.show_user) { this.show_user = json.show_user; }
			if (json.address_id) { this.address_id = json.address_id; }
			if (json.address) { this.address = new Address(json.address); }
			if (json.provider_id) { this.provider_id = json.provider_id; }
			if (json.provider) { this.provider = new Provider(json.provider); }
			if (json.category_id) { this.category_id = json.category_id; }
			if (json.category) { this.category = new Category(json.category); }
			if (json.schedules && json.schedules.length) { this.schedules = this.buildScheduleArray(json.schedules); }
			if (json.tags) { this.tags = json.tags; }
			if (json.target_groups) { this.target_groups = json.target_groups; }
			// this.translations = Object.assign(json._translations);
		}
	}

	public name: string = '';
	public description: string = '';
	public show_user: boolean = false;

	public address_id: string = '';
	public address: Address = new Address({});
	public provider_id: string = '';
	public provider: Provider = new Provider({});
	public category_id: string = '';
	public category: Category = new Category();

	public schedules: Schedule[] = [];
	public tags: Tag[] = [];
	public target_groups: TargetGroup[] = [];
	public translations: any = {};

	buildScheduleArray(dates: any[]): Schedule[] {
		const scheduleArray: Schedule[] = [];
		for (const date of dates) {
			scheduleArray.push(new Schedule(date));
		}
		return scheduleArray;
	}

}

File: server/wupportal/src/main/java/de/codeschluss/wupportal/model/Activity.java
package de.codeschluss.wupportal.model;

import java.io.Serializable;
import java.util.List;

import javax.persistence.*;

/**
 * The persistent class for the activities database table.
 * 
 */
@Entity
@Table(name="activities")
public class Activity extends BaseEntity implements Serializable {

	private static final long serialVersionUID = 1L;

	private String name;
	
	@Lob
	private String description;

	@Column(name="show_user")
	private boolean showUser;

	@ManyToOne
	private Address address;

	@ManyToOne
	private Category category;

	@ManyToOne
	private Provider provider;

	@ManyToMany()
	@JoinTable(
			name = "activities_tags",
			joinColumns = @JoinColumn(name = "activity_id"),
			inverseJoinColumns = @JoinColumn(name = "tag_id"))
	private List<Tag> tags;

	@ManyToMany()
	@JoinTable(
			name = "activities_target_groups",
			joinColumns = @JoinColumn(name = "activity_id"),
			inverseJoinColumns = @JoinColumn(name = "target_group_id"))
	private List<TargetGroup> targetGroups;
	
	@OneToMany(mappedBy = "activity")
	private List<Schedule> schedules;

	public Activity() {
	}

	public String getDescription() {
		return this.description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public String getName() {
		return this.name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public boolean getShowUser() {
		return this.showUser;
	}

	public void setShowUser(boolean showUser) {
		this.showUser = showUser;
	}

	public Address getAddress() {
		return this.address;
	}

	public void setAddress(Address address) {
		this.address = address;
	}

	public Category getCategory() {
		return this.category;
	}

	public void setCategory(Category category) {
		this.category = category;
	}

	public Provider getProvider() {
		return this.provider;
	}

	public void setProvider(Provider provider) {
		this.provider = provider;
	}

	public List<Tag> getTags() {
		return this.tags;
	}

	public void setTags(List<Tag> tags) {
		this.tags = tags;
	}

	public List<TargetGroup> getTargetGroups() {
		return this.targetGroups;
	}

	public void setTargetGroups(List<TargetGroup> targetGroups) {
		this.targetGroups = targetGroups;
	}
	
	public List<Schedule> getSchedules() {
		return this.schedules;
	}

	public void setSchedules(List<Schedule> schedules) {
		this.schedules = schedules;
	}

}